fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostCalculator.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.doris.nereids.operators.plans.physical.PhysicalHashJoin;
+import org.apache.doris.nereids.operators.plans.physical.PhysicalOlapScan;
+import org.apache.doris.nereids.operators.plans.physical.PhysicalProject;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.statistics.StatsDeriveResult;
+
+import com.google.common.base.Preconditions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Calculate the cost of a plan.
+ * Inspired by Presto.
+ */
+public class CostCalculator {
+    /**
+     * Constructor.
+     */
+    public double calculateCost(GroupExpression groupExpression) {
+        PlanContext planContext = new PlanContext(groupExpression);
+        CostEstimator costCalculator = new CostEstimator();
+        CostEstimate costEstimate = groupExpression.getOperator()
+                .accept(costCalculator, planContext);
+        return costFormula(costEstimate);
+    }
+
+    private double costFormula(CostEstimate costEstimate) {
+        double cpuCostWeight = 1;
+        double memoryCostWeight = 1;
+        double networkCostWeight = 1;
+        return costEstimate.getCpuCost() * cpuCostWeight + costEstimate.getMemoryCost() * memoryCostWeight
+                + costEstimate.getNetworkCost() * networkCostWeight;
+    }
+
+    private static class CostEstimator extends OperatorVisitor<CostEstimate, PlanContext> {
+        @Override
+        public CostEstimate visitOperator(Operator operator, PlanContext context) {
+            return CostEstimate.zero();
+        }
+
+        @Override
+        public CostEstimate visitPhysicalAggregation(PhysicalAggregation physicalAggregation, PlanContext context) {
+            StatsDeriveResult statistics = context.getStatisticsWithCheck();
+            return CostEstimate.ofCpu(statistics.computeSize());
+        }
+
+        @Override
+        public CostEstimate visitPhysicalOlapScan(PhysicalOlapScan physicalOlapScan, PlanContext context) {
+            StatsDeriveResult statistics = context.getStatisticsWithCheck();
+            return CostEstimate.ofCpu(statistics.computeSize());
+        }
+
+        @Override
+        public CostEstimate visitPhysicalHashJoin(PhysicalHashJoin physicalHashJoin, PlanContext context) {
+            Preconditions.checkState(context.getGroupExpression().arity() == 2);
+            Preconditions.checkState(context.getChildrenStats().size() == 2);
+
+            StatsDeriveResult leftStatistics = context.getChildStatistics(0);
+            StatsDeriveResult rightStatistics = context.getChildStatistics(1);
+
+            // TODO: handle some case
+
+            List<SlotId> leftIds = context.getChildOutputSlots(0).stream().map(Slot::getId).map(SlotId::create)
+                    .collect(Collectors.toList());
+            List<SlotId> rightIds = context.getChildOutputSlots(1).stream().map(Slot::getId).map(SlotId::create)
```

Comment:
StatsDeriveResult will be modified to adjust for Nereids. After that, it could use both SlotId and ExprId as Column key